src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -326,4 +327,13 @@ public long weightedSize()
                 .map(policy -> policy.weightedSize().orElseGet(cache::estimatedSize))
                 .orElseGet(cache::estimatedSize);
     }
+
+    /**
+     * Returns the number of cached chunks of given file.
+     */
+    @VisibleForTesting
+    public int sizeOfFile(String filePath)
+    {
+        return (int) cache.asMap().keySet().stream().filter(x -> x.path.equals(filePath)).count();
```

Comment:
Depending on this being on a hot path or not we might contemplate about removing streams in favor of plain for loop.

https://lists.apache.org/thread/65glsjzkmpktzmns6j9wvr4nczvskx36